### PR TITLE
amitha/feature/dialog-is-reference-hidden

### PR DESCRIPTION
**Pull request by amit-hanoch**
Hide dialog when reference is hidden

#### Basic
- [ ] PR has description
- [ ] Changes to the component are backward compatible (including selectors structure). If not - add to the title of the PR "BREAKABLE_CHANGE""
- [ ] All changes to the component are reflected in the ReadMe
- [ ] If component is old and was not compliant with the latest guidelines - it was fixed (optional) 
#### Style
- [ ] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [ ] All the changes to the component should be reflected in the Storybook.
- [ ] Component passed Accessibility Plugin checks
#### Tests
- [ ] All current tests are passing
- [ ] New functionality is covered with tests
- [ ] Tests are compliant with TESTING_README.md instructions

### CODE REVIEW

**Review by orrgottlieb — APPROVED** [review]

File: src/components/Dialog/Dialog.jsx [hunk 1]
```diff
@@ -346,12 +348,19 @@ export default class Dialog extends PureComponent {
...
               }
+
+              if (hideWhenReferenceHidden && isReferenceHidden) {
+                const event = new CustomEvent("onReferenceHidden");
+                this.hideDialog(event, "onReferenceHidden");
+              }
+
               return (
                 <DialogContent
+                  isReferenceHidden={hideWhenReferenceHidden && isReferenceHidden}
```
Comment (MosheZemah) [inline src/components/Dialog/Dialog.jsx:363]: maybe send isReferenceHidden as is
and also pass the other prop (hideWhenReferenceHidden) to the dialogContent component?

File: src/components/Dialog/Dialog.jsx [hunk 2]
```diff
@@ -346,12 +348,19 @@ export default class Dialog extends PureComponent {
               ...modifiers
             ]}
           >
-            {({ placement, style, ref, arrowProps }) => {
+            {({ placement, style, ref, arrowProps, isReferenceHidden }) => {
```
Comment (MosheZemah) [inline src/components/Dialog/Dialog.jsx:351]: @orrgottlieb are we ok with sending new objects/arrays every time to the popper component?

Comment (orrgottlieb) [reply src/components/Dialog/Dialog.jsx:351]: it is the render props methodology, it is a render function. isReferenceHidden is a boolean @MosheZemah

Comment (MosheZemah) [reply src/components/Dialog/Dialog.jsx:351]: I meant the code above that line (that wasn't changed)

**Pull request merged**